File: example/typescript/main/TypesModule/Api/Properties.ts
namespace TypesModule {
    export class DateRangeProperty {
        private from = STRING
        private to = STRING

        static create(
            from: string,
            to: string,
        ): DateRangeProperty {
            const instance = new DateRangeProperty()
            instance.from = from
            instance.to = to
            return instance
        }

        getFrom(): Date {
            return CustomTypesMapper.dateCreate(this.from)
        }

        getTo(): Date {
            return CustomTypesMapper.dateCreate(this.to)
        }
    }
}

File: example/kotlin/src/main/kotlin/com/some/pkg/typesmodule/api/Properties.kt
package com.some.pkg.typesmodule.api

data class DateRangeProperty(
    private val from: String,
    private val to: String,
) {
    fun getFrom(): Date {
        return dateCreate(this.from)
    }

    fun getTo(): Date {
        return dateCreate(this.to)
    }
}